lib/longleaf/helpers/s3_uri_helper.rb

```diff
@@ -31,6 +31,30 @@ def self.extract_bucket(url)
         return prefix[0..-2]
       end
     end
+    
+    def self.extract_path(url)
+      uri = s3_uri(url)
+
+      matches = ENDPOINT_PATTERN.match(uri.host)
+      if matches.nil?
+        raise ArgumentError.new("Provided URI does match the expected pattern for an S3 URI")
+      end
+
+      path = uri.path
+      return nil if path == '/'
+      
+      # trim off the first slash
+      path = path.partition('/').last
+      
+      # Determine if the first part of the path is the bucket name
+      prefix = matches[1]
+      if prefix.nil? || prefix.empty?
+        # trim off the bucket name
+        path = path.partition('/').last
```

Comment:
So if the bucket is not in the host string, assume the parth starts with the bucket?

Reply:
Right, this is one of the acceptable formats for a bucket uri
`http://s3-my-region-amazonaws.com/bucketname/`
while the other acceptable format is:
`http://bucketname.s3-my-region-amazonaws.com/` (with a few minor variations)

The bucket has to be present in the path uri.